code/midtown/mmai/aiVehicleOpponent.cpp

```diff
@@ -20,5 +20,56 @@ define_dummy_symbol(mmai_aiVehicleOpponent);
 
 #include "aiVehicleOpponent.h"
 
+#include "agi/dlptmpl.h"
+#include "agi/getdlp.h"
+#include "agiworld/quality.h"
+
+#include "aiData.h"
+#include "aiGoalBackup.h"
+#include "aiGoalFollowWayPts.h"
+#include "aiGoalStop.h"
+
 void aiVehicleOpponent::DrawDamage()
 {}
+
+void aiVehicleOpponent::Init(i32 opp_id, aiRaceData* race_data, char* race_name)
+{
+    i32 texture_quality = agiRQ.TextureQuality;
+    if (agiRQ.TextureQuality)
+        --agiRQ.TextureQuality;
+
+    i32 paint_job = opp_id & 3;
+    i32 index = opp_id + 1;
+
+    OpponentRaceData* oppRaceData = static_cast<OpponentRaceData*>(race_data->Opponents.Access(index));
+
+    Car.Init(oppRaceData->Model, CAR_TYPE_OPPONENT, paint_job);
+
+    agiRQ.TextureQuality = texture_quality;
+
+    aiVehicle::Init(opp_id);
+    Car.Reset();
+
+    WayPts = arnew aiGoalFollowWayPts(oppRaceData->PathFile, &RailSet, this, &IsBackup, &IsFinished, &IsStopped,
+        xconst(race_name), oppRaceData->MaxThrottle);
+
+    BackupGoal = arnew aiGoalBackup(&RailSet, &Car, &IsBackup);
+
+    StopGoal = arnew aiGoalStop(&Car, &IsStopped);
+
+    IsSemi = !std::strcmp("vpsemi", oppRaceData->Model);
+
+    DLPTemplate* DLPTemplate = GetDLPTemplate(oppRaceData->Model);
```

Comment:
Use snake_case variable names (and try to avoid naming a variable the same as a type)